install-tests/react-native/cli.ts

```diff
@@ -0,0 +1,263 @@
+    (args) =>
+      args
+        .option("realm-version", { type: "string", default: "latest" })
+        .option("react-native-version", { type: "string", default: "latest" })
+        .option("engine", { type: "string", choices: ["hermes", "jsc"], default: "hermes" })
+        .option("force", { description: "Delete any existing app directory", type: "boolean", default: false })
+        .option("skip-bundle-install", {
+          description: "Skip the iOS specific 'bundle install'",
+          type: "boolean",
+          default: false,
+        })
+        .option("skip-pod-install", {
+          description: "Skip the iOS specific 'pod install'",
+          type: "boolean",
+          default: false,
+        }),
+    (argv) => {
+      const {
+        "app-path": appPath,
+        "realm-version": realmVersion,
+        "react-native-version": reactNativeVersion,
+        "new-architecture": newArchitecture,
+        "skip-bundle-install": skipBundleInstall,
+        "skip-pod-install": skipPodInstall,
+        engine,
+        force,
+      } = argv;
+
+      const env = getEnv(newArchitecture);
+
+      console.log(`Initializing react-native@${reactNativeVersion} template into '${appPath}'`);
+      console.log("New achitecture is", newArchitecture ? "enabled" : "disabled");
+
+      if (fs.existsSync(appPath) && force) {
+        console.log("Deleting existing app directory! (because --force)");
+        fs.rmSync(appPath, { recursive: true });
+      }
+
+      exec("npx", [
+        "--yes",
+        "react-native",
+        "init",
+        "--npm",
+        "--skip-install", // We'll do this in a different step
+        appName,
+        "--version",
+        reactNativeVersion,
+        "--directory",
+        appPath,
+      ]);
+
+      console.log(`Adding realm@${realmVersion} to the app (and installing dependencies)`);
+      // We're using force to succeed on peer dependency issues
+      exec("npm", ["install", `realm@${realmVersion}`, "--force"], { cwd: appPath });
+
+      const podfilePath = path.resolve(appPath, "ios", "Podfile");
+      console.log(`Patching podfile to use ccache (${podfilePath})`);
+      exec("patch", [podfilePath, CCACHE_PODFILE_PATCH_PATH]);
+
+      if (engine === "jsc") {
+        console.log(`Patching Podfile to use JSC (${podfilePath})`);
+        exec("patch", [podfilePath, JSC_PODFILE_PATCH_PATH]);
+
+        const appGradleBuildPath = path.resolve(appPath, "android", "app", "build.gradle");
+        console.log(`Patching app/build.gradle to use JSC (${appGradleBuildPath})`);
+        exec("patch", [appGradleBuildPath, JSC_BUILD_GRADLE_PATCH_PATH]);
+      }
+
+      if (!skipBundleInstall) {
+        console.log(`Installing gem bundle (needed to pod-install for iOS)`);
+        exec("bundle", ["install"], { cwd: appPath });
+      }
+
+      if (!skipPodInstall) {
+        console.log(`Installing CocoaPods`);
+        exec("pod", ["install"], { cwd: path.resolve(appPath, "ios"), env });
```

Comment:
Maybe not an issue, but `npx pod-install` will automatically call `repo update` if this fails.

Reply:
Let's add this for good measures then.

```suggestion
        // Use --no-repo-update to avoid updating the repo if the install fails
        exec("pod", ["install", "--no-repo-update"], { cwd: path.resolve(appPath, "ios"), env });
```